### PR TITLE
Add blog author detail page

### PR DESCRIPTION
**Pull request by lukeocodes**

### CODE REVIEW

File: blog/src/components/authors/AuthorHero.astro [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
+import Svg from "../../shared/components/general/Svg.astro";
+import Icon from "../../shared/components/general/Icon.astro";
 
 const { author } = Astro.props;
 ---
 
 <PrimarySection class="bg-darkCharcoal">
-	<div class="author-hero">
+	<a href="/authors" class="col-span-2 xl:col-span-3 inline-block py-4 md:py-6 lg:py-8 xl:py-10">&lt;- All authors</a>
+	<div class="author-hero mb-20">
```
Comment (SandraRodgers) [inline blog/src/components/authors/AuthorHero.astro:11]: On mobile xs size, the design shows spacing between the author picture and the link "All authors". There also should be a little spacing between the picture and the author name below.

File: blog/src/components/authors/AuthorPostsPage.astro [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import BlogCard from "../../shared/components/cards/BlogCard.astro";
+import Pagination from "../nav/Pagination.astro";
+
+const { posts, page, slug, path } = Astro.props;
+---
+
+<div class="w-full">
+	<div class="relative mx-auto">
+		<div class="grid grid-flow-row-dense grid-cols-1 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 justify-items-center gap-4 lg:gap-6">
+			{posts.map((post, index) => <BlogCard post={post} />)}
```
Comment (SandraRodgers) [inline blog/src/components/authors/AuthorPostsPage.astro:12]: Sort by date

File: blog/src/components/authors/Hero.astro [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import RightDecoration from "../../shared/components/decoration/RightDecoration.astro";
+import Svg from "../../shared/components/general/Svg.astro";
+import PrimarySection from "../../shared/components/layout/PrimarySection.astro";
+
+const { title } = Astro.props;
+---
+
+<PrimarySection class="">
+	<div class="grid grid-cols-2 xl:grid-cols-3">
+		<a href="/posts" class="col-span-2 xl:col-span-3 inline-block py-4 md:py-6 lg:py-8 xl:py-10">&lt;- All posts</a>
```
Comment (SandraRodgers) [inline blog/src/components/authors/Hero.astro:12]: mobile design shows 44 px between 'All posts' link and 'Authors' title, so we might need to increase this padding a bit

File: blog/src/pages/authors/[author]/[...page].astro [hunk 4]
```diff
@@ -21,27 +23,33 @@ export async function getStaticPaths({ paginate }) {
 
 	return authors.map((author) => {
 		const posts = allPosts.filter((post) => {
-			return post.frontmatter.category === author.slug;
+			if (post.frontmatter.authors) {
```
Comment (SandraRodgers) [inline blog/src/pages/authors/[author]/[...page].astro:26]: Sort by date - newest first

**Review by phazonoverload — APPROVED** [review]

**Pull request merged**